### DISCUSSION TITLE
qlight block sync Error

### DISCUSSION THREAD

**Discussion by vata500** [ConsenSys/quorum#1610] [Q&A]
I'm testing seven qbft quorum nodes. However, the problem is that attaching qlight fails to sync the block. I think I've done all the necessary settings, so what's the problem?

Below is the execution log of qlight geth. Thank you.

/ INFO [02-21|09:46:24.683] Maximum peer count                       ETH=50 LES=0 total=50
WARN [02-21|09:46:24.684] The flag --rpc is deprecated and will be removed in the future, please use --http 
WARN [02-21|09:46:24.684] The flag --rpcaddr is deprecated and will be removed in the future, please use --http.addr 
WARN [02-21|09:46:24.684] The flag --rpcport is deprecated and will be removed in the future, please use --http.port 
WARN [02-21|09:46:24.684] The flag --rpccorsdomain is deprecated and will be removed in the future, please use --http.corsdomain 
WARN [02-21|09:46:24.684] The flag --rpcapi is deprecated and will be removed in the future, please use --http.api 
WARN [02-21|09:46:24.684] The flag --rpcvhosts is deprecated and will be removed in the future, please use --http.vhosts 
INFO [02-21|09:46:24.684] Smartcard socket not found, disabling    err="stat /run/pcscd/pcscd.comm: no such file or directory"
WARN [02-21|09:46:24.684] Option nousb is deprecated and USB is deactivated by default. Use --usb to enable 
INFO [02-21|09:46:24.684] The node is configured to run as a qlight client. 'maxpeers' is overridden to `1` and the P2P listener is disabled. 
DEBUG[02-21|09:46:24.684] FS scan times                            list="33.198µs" set="3.683µs" diff="2.213µs"
TRACE[02-21|09:46:24.684] Started watching keystore folder         path=/phnx/data/keystore
DEBUG[02-21|09:46:24.684] Failed to decode keystore key            path=/phnx/data/keystore/accountPassword err=EOF
DEBUG[02-21|09:46:24.684] Failed to decode keystore key            path=/phnx/data/keystore/accountPrivateKey err="json: cannot unmarshal number into Go value of type struct { Address string \"json:\\\"address\\\"\" }"
DEBUG[02-21|09:46:24.684] Failed to decode keystore key            path=/phnx/data/keystore/accountAddress    err="json: cannot unmarshal number into Go value of type struct { Address string \"json:\\\"address\\\"\" }"
TRACE[02-21|09:46:24.684] Handled keystore changes                 time="92.915µs"
WARN [02-21|09:46:24.684] Using deprecated resource file /phnx/data/nodekey, please move this file to the 'geth' subdirectory of datadir. 
DEBUG[02-21|09:46:24.684] Sanitizing Go's GC trigger               percent=100
WARN [02-21|09:46:24.684] WARNING: The flag --istanbul.blockperiod is deprecated and will be removed in the future, please use ibft.blockperiodseconds on genesis file 
INFO [02-21|09:46:24.684] Enabling recording of key preimages since archive mode is used 
INFO [02-21|09:46:24.684] Set global gas cap                       cap=25,000,000
INFO [02-21|09:46:24.684] Allocated trie memory caches             clean=154.00MiB dirty=256.00MiB
INFO [02-21|09:46:24.684] Allocated cache and file handles         database=/phnx/data/geth/chaindata cache=768.00MiB handles=524,288
DEBUG[02-21|09:46:24.703] Chain freezer table opened               database=/phnx/data/geth/chaindata/ancient table=headers items=0 size=0.00B
DEBUG[02-21|09:46:24.705] Chain freezer table opened               database=/phnx/data/geth/chaindata/ancient table=hashes  items=0 size=0.00B
DEBUG[02-21|09:46:24.706] Chain freezer table opened               database=/phnx/data/geth/chaindata/ancient table=bodies  items=0 size=0.00B
DEBUG[02-21|09:46:24.774] Chain freezer table opened               database=/phnx/data/geth/chaindata/ancient table=receipts items=0 size=0.00B
DEBUG[02-21|09:46:24.776] Chain freezer table opened               database=/phnx/data/geth/chaindata/ancient table=diffs    items=0 size=0.00B
INFO [02-21|09:46:24.776] Opened ancient database                  database=/phnx/data/geth/chaindata/ancient readonly=false
DEBUG[02-21|09:46:24.776] Current full block not old enough        number=0 hash=85b43b..69178f delay=3,162,240
DEBUG[02-21|09:46:24.776] Account Extra Data root                  hash=000000..000000
INFO [02-21|09:46:24.776] Initialised chain configuration          config="{ChainID: 1337 Homestead: 0 DAO: <nil> DAOSupport: false EIP150: 0 EIP155: 0 EIP158: 0 Byzantium: 0 IsQuorum: true Constantinople: 0 TransactionSizeLimit: 64 MaxCodeSize: 0 Petersburg: <nil> Istanbul: <nil>, Muir Glacier: <nil>, Berlin: <nil>  Catalyst: <nil> YOLO v3: <nil> PrivacyEnhancements: <nil> PrivacyPrecompile: <nil> EnableGasPriceBlock: <nil> Engine: unknown}"
INFO [02-21|09:46:24.777] Initialising Ethereum protocol           network=1337 dbversion=<nil>
INFO [02-21|09:46:24.777] Initialising Quorum consensus protocol   name=istanbul versions="[100 99 64]" network=1337 dbversion=<nil>
INFO [02-21|09:46:24.777] Using SegregatedCacheProvider. 
DEBUG[02-21|09:46:24.777] Account Extra Data root                  hash=000000..000000
INFO [02-21|09:46:24.777] Loaded most recent local header          number=0 hash=85b43b..69178f td=1 age=5y11mo1w
INFO [02-21|09:46:24.777] Loaded most recent local full block      number=0 hash=85b43b..69178f td=1 age=5y11mo1w
INFO [02-21|09:46:24.777] Loaded most recent local fast block      number=0 hash=85b43b..69178f td=1 age=5y11mo1w
DEBUG[02-21|09:46:24.778] Account Extra Data root                  hash=000000..000000
DEBUG[02-21|09:46:24.778] Account Extra Data root                  hash=000000..000000
TRACE[02-21|09:46:24.778] header data not found in ancient or level db hash=000000..000000
WARN [02-21|09:46:24.778] Failed to load snapshot, regenerating    err="missing or corrupted snapshot"
INFO [02-21|09:46:24.778] Rebuilding state snapshot 
DEBUG[02-21|09:46:24.778] Journalled generator progress            progress=empty
DEBUG[02-21|09:46:24.778] Start snapshot generation                root=1e239f..e3e8fe
DEBUG[02-21|09:46:24.778] Account Extra Data root                  hash=000000..000000
DEBUG[02-21|09:46:24.778] Account Extra Data root                  hash=000000..000000
DEBUG[02-21|09:46:24.778] Reinjecting stale transactions           count=0
INFO [02-21|09:46:24.778] Resuming state snapshot generation       root=1e239f..e3e8fe accounts=0 slots=0 storage=0.00B elapsed="215.246µs"
INFO [02-21|09:46:24.778] Regenerated local transaction journal    transactions=0 accounts=0
TRACE[02-21|09:46:24.778] Detected outdated state range            kind=account prefix=0x61 last=0x err="wrong root: have 0x56e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421 want 0x1e239f73565e814f0100214a315e07aec02750ded4587cd709cfacd8b7e3e8fe"
DEBUG[02-21|09:46:24.778] Recalculated downloader QoS values       rtt=20s confidence=1.000 ttl=1m0s
DEBUG[02-21|09:46:24.778] Regenerated state range                  kind=account prefix=0x61 root=1e239f..e3e8fe last=0x count=7 created=7 updated=0 untouched=0 deleted=0
DEBUG[02-21|09:46:24.778] Journalled generator progress            progress=done
INFO [02-21|09:46:24.778] BFT: Initialising snap with extradata    validators="[0x1dFe1b04F31a2feB176998a6EBDd98eDda18F8C4 0xC3D6b700B7C636C8285f33E13BC3626271Ea4023 0x829819ca7beA981Ed05fe84BcCE0274588fB4d77 0xFd8c0a2555a05adDed8aF7CD6b0631a66b256129 0xa82bDd94B9f5247B59AC218bFD5f47cbba3b55a0 0x92922fBbaE24f4B915F5B0187a342BFcD561e463 0x5802c0Baf69131d1f2b4D8c23d0fDf8867EFcdD8]"
INFO [02-21|09:46:24.778] Generated state snapshot                 accounts=7 slots=0 storage=350.00B elapsed="623.298µs"
DEBUG[02-21|09:46:24.778] BFT: store snapshot to database          snap.number=0 snap.hash=0x85b43b1ad0e450b70f35883c343fd0656aae63b637ae2163807b52bb4d69178f snap.epoch=30000 snap.validators="[0x1dFe1b04F31a2feB176998a6EBDd98eDda18F8C4 0x5802c0Baf69131d1f2b4D8c23d0fDf8867EFcdD8 0x829819ca7beA981Ed05fe84BcCE0274588fB4d77 0x92922fBbaE24f4B915F5B0187a342BFcD561e463 0xa82bDd94B9f5247B59AC218bFD5f47cbba3b55a0 0xC3D6b700B7C636C8285f33E13BC3626271Ea4023 0xFd8c0a2555a05adDed8aF7CD6b0631a66b256129]" snap.votes=[]
WARN [02-21|09:46:24.778] Error reading unclean shutdown markers   error="leveldb: not found"
WARN [02-21|09:46:24.778] permission-config.json file is missing. Smart-contract-based permission service will be disabled error="stat data/permission-config.json: no such file or directory"
INFO [02-21|09:46:24.778] extension service registered 
INFO [02-21|09:46:24.779] No plugins to initialise 
INFO [02-21|09:46:24.779] Starting peer-to-peer node               instance=Geth/node/v1.10.3-stable-93c5f781(quorum-v22.7.5)/linux-amd64/go1.16.15
DEBUG[02-21|09:46:24.779] Account Extra Data root                  hash=000000..000000
DEBUG[02-21|09:46:24.779] Account Extra Data root                  hash=000000..000000
TRACE[02-21|09:46:24.779] header data not found in ancient or level db hash=000000..000000
TRACE[02-21|09:46:24.786] Adding static node                       id=872ca7fd5f2c62e3 ip=10.10.0.38 added=true
INFO [02-21|09:46:24.787] New local node record                    seq=1 id=872ca7fd5f2c62e3 ip=127.0.0.1  udp=0 tcp=0
INFO [02-21|09:46:24.787] Started P2P networking                   self=enode://a3b56531e1715f4d2d474118a00109910394d39c260744f0bdae44367b401a04c7a44d545ba1781fb5cc845b3ff6dfbba367af4a0f847ddc038f97455cbfe1fc@127.0.0.1:0
DEBUG[02-21|09:46:24.788] IPCs registered                          namespaces=admin,debug,web3,eth,txpool,personal,istanbul,miner,net,quorumExtension
INFO [02-21|09:46:24.788] IPC endpoint opened                      url=/phnx/data/geth.ipc isMultitenant=false
INFO [02-21|09:46:24.788] Security Plugin is not enabled 
ERROR[02-21|09:46:24.788] Unavailable modules in HTTP API list     unavailable="[shh quorum]" available="[admin debug web3 eth txpool personal istanbul miner net quorumExtension]"
INFO [02-21|09:46:24.788] Security: TLS not enabled                endpoint=0.0.0.0:8545 reason="no TLSConfigurationSource found"
INFO [02-21|09:46:24.788] HTTP server started                      endpoint=[::]:8545    prefix=     cors=* vhosts=* isTls=false
DEBUG[02-21|09:46:24.788] extension service: starting 
DEBUG[02-21|09:46:24.788] Enrich call context with values from security context 
DEBUG[02-21|09:46:24.788] Served eth_subscribe                     reqid=private/1 t="100.017µs"
TRACE[02-21|09:46:24.788] Handled RPC response                     reqid=private/1 t="2.321µs"
DEBUG[02-21|09:46:24.789] Enrich call context with values from security context 
DEBUG[02-21|09:46:24.789] Served eth_subscribe                     reqid=private/1 t="72.722µs"
TRACE[02-21|09:46:24.789] Handled RPC response                     reqid=private/1 t="2.262µs"
DEBUG[02-21|09:46:24.789] Enrich call context with values from security context 
DEBUG[02-21|09:46:24.789] Served eth_subscribe                     reqid=private/1 t="55.739µs"
TRACE[02-21|09:46:24.789] Handled RPC response                     reqid=private/1 t="1.838µs"
WARN [02-21|09:46:24.789] ------------------------------------------------------------------- 
WARN [02-21|09:46:24.789] Referring to accounts by order in the keystore folder is dangerous! 
WARN [02-21|09:46:24.789] This functionality is deprecated and will be removed in the future! 
WARN [02-21|09:46:24.789] Please use explicit addresses! (can search via `geth account list`) 
WARN [02-21|09:46:24.789] ------------------------------------------------------------------- 
INFO [02-21|09:46:25.487] Unlocked account                         address=0x5083Aa27E6acbC7E37743bBD7176aa84951412e2
WARN [02-21|09:46:25.487] permission-config.json file is missing. Smart-contract-based permission service will be disabled error="stat /phnx/data/permission-config.json: no such file or directory" /

**Comment by baptiste-b-pegasys** [comment]
qlight is a secure proxy node that connects to a qlight server, it doesn't sync, everything is done in qlight server
